### ISSUE TITLE
Provide sample programs for PSA APIs

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#9416]
Provide sample programs for PSA Crypto APIs.

The scope isn't defined precisely: we don't need sample code for every single function, but we need the most common use cases.

Hannes has already written PSA sample programs in https://github.com/Mbed-TLS/mbedtls/pull/5064. We need to upstream them as has been done for the hash program in https://github.com/Mbed-TLS/mbedtls/pull/7942.

The programs should be added to all branches where the relevant functions exist, not just to `development`, since the lack of sample programs is a long-standing defect that 4.0 is just making more acute.

**Comment by davidhorstmann-arm** [comment]
A note that this was previously work in progress by @tom-daubney-arm

**Comment by tom-daubney-arm** [comment]
I have split out #5064 into separate PRs and I have progressed them all locally and pushed some for review. Currently https://github.com/Mbed-TLS/mbedtls/pull/7954 is waiting on review but has not been able to be prioritised yet due to other work.

The rest of the PRs are:
#7953
#7951 
#7948 
#7944
#7938 

Note that a small amount of work is needed to happen on the CI testing side for the example programs. Currently it is not possible to specify an "anti-dependency" i.e. a config option that needs to be switched off in order for the program to run. I have implemented a solution locally (not pushed anywhere) that allows us to specify anti-dependencies for the program in the CI scripts by prepending an "!" to the config option name, like we do in the main test framework. I can raise a separate PR for this.
